### DISCUSSION TITLE
How to get GPU acceleration while calculating embeddings using a local model?

### DISCUSSION THREAD

**Discussion by siddhsql** [langchain4j/langchain4j#1057] [Q&A]
Hello,

I am synced to version 0.22.0 of langchain4j and using the `dev.langchain4j.model.inprocess.InProcessEmbeddingModelType.ALL_MINILM_L6_V2` to calculate embeddings on a M2 Mac Mini. It does not seem to use the GPU and as a result is a bit slow. How can I get GPU acceleration? Thanks very much.

**Comment by siddhsql** [comment]
Hi All, thanks for all your inputs. once thing i wanted to understand (and maybe this is topic for separate thread but i will start it here), is that when i looked at the dependency graph of langchain4j i see it uses both the dlj library from amazon and the onnxruntime from microsoft. i am not familiar with internals of any of these libraries but wouldn't it be better for us to stick with one deep learning library? am i understanding something wrong here?

**Reply by Craigacp** [reply]
DJL wraps ONNX Runtime among other backends.

**Comment by siddhsql** [comment]
I now tried to run the same code on a machine with NVIDIA RTX A6000 GPU. again it seems by default langchain4j does not use the GPU if available. What can I do to make it use GPU in this case? My code:

```
import static dev.langchain4j.model.inprocess.InProcessEmbeddingModelType.ALL_MINILM_L6_V2;
...
model = new InProcessEmbeddingModel(ALL_MINILM_L6_V2);
model.embed("some text whose embeddings you want to calculate").vector()
```

I am only showing minimal code. I calculated thousands of embeddings this way and GPU usage was 0. CPU was maxed out.

**Comment by siddhsql** [comment]
thanks for the response. a separate question - how can I get the source
code of version 0.22.0 of dev.langchain4j:langchain4j-embeddings? that is
the one i'd like to use. the repo does not have that tag anymore.

**Comment by siddhsql** [comment]
thanks a lot.
